mmpretrain/models/utils/swiglu_ffn.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from .layer_scale import LayerScale
+
+
+class SwiGLUFFN(nn.Module):
+    """SwiGLU FFN layer.
+
+    Modified from https://github.com/facebookresearch/dinov2/blob/main/dinov2/layers/swiglu_ffn.py
+    """  # noqa
+
+    def __init__(
+        self,
+        embed_dims: int,
+        feedforward_channels: Optional[int] = None,
+        out_dims: Optional[int] = None,
+        layer_scale_init_value: float = 0.,
+        bias: bool = True,
```

Comment:
It is possible to add an `add_identity` to decide whether to add identity or not , `dropout_layer` or `droprate` to add a dropout layer at the end. Beside, in EVA02, there is a sub_layernorm before w3, this function could be added.

Reply:
updated, please check it